rotkehlchen/chain/ethereum/transactions.py

```diff
@@ -16,3 +28,71 @@ def __init__(
             database: 'DBHandler',
     ) -> None:
         super().__init__(evm_inquirer=ethereum_inquirer, database=database)
+
+    def query_for_graph_delegation_txns(self, addresses: Sequence[ChecksumEvmAddress]) -> None:
+        if len(addresses) == 0:
+            return None
+
+        staking_contract = self.evm_inquirer.contracts.contract(CONTRACT_STAKING)
+        for address in addresses:
+            # fetch contract approval events to consider if we query the logs
+            db_filter = EvmEventFilterQuery.make(
+                counterparties=[CPT_THEGRAPH],
+                location=Location.from_chain_id(self.evm_inquirer.chain_id),
+                event_types=[HistoryEventType.INFORMATIONAL],
+                event_subtypes=[HistoryEventSubType.APPROVE],
+            )
+            dbevents = DBHistoryEvents(self.database)
+            with dbevents.db.conn.read_ctx() as cursor:
+                if len(dbevents.get_history_events(
+                        cursor=cursor,
+                        filter_query=db_filter,
+                        has_premium=True,
+                )) == 0:
+                    continue
+
+            with self.database.conn.read_ctx() as cursor:
+                from_block = staking_contract.deployed_block
+                if (result := self.database.get_dynamic_cache(
+                    cursor=cursor,
+                    name=DBCacheDynamic.LAST_BLOCK_ID,
+                    location=self.evm_inquirer.chain_name,
+                    location_name=LAST_GRAPH_DELGATIONS,
+                    account_id=address,
+                )) is not None:
+                    from_block = result
+
+            to_block = self.evm_inquirer.get_latest_block_number()
+            argument_filters = {
+                'l2Delegator': address,
+            }
+            events = self.evm_inquirer.get_logs(
+                contract_address=staking_contract.address,
+                abi=staking_contract.abi,
+                event_name='DelegationTransferredToL2',
+                argument_filters=argument_filters,
+                from_block=from_block,
+                to_block=to_block,
+            )
+            if len(events) == 0:
+                continue
```

Comment:
shouldn't you save that you queried until this block? You seem to set it at the end of the loop. I would put it in a if/else

Reply:
Added